File: 1635-number-of-good-pairs/number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        num_dict = defaultdict(int)
        for num in nums:
            num_dict[num] += 1
        
        total_count = 0

        for num in num_dict:
            count = num_dict[num]
            if count >= 2:
                total_count += count * (count -1) // 2
            
        return total_count

File: 1635-number-of-good-pairs/number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> num_map = new HashMap<>();

        int total_count = 0;

        for (int num: nums) {
            int pre_count = num_map.getOrDefault(num, 0);

            total_count += pre_count;

            num_map.put(num, pre_count+1);
        }

        return total_count;
    }
}